trl/trainer/accelerate_ppo.py

```diff
@@ -0,0 +1,363 @@
+
+
+    def step(self, queries, responses, scores):
+        """
+        Run a PPO optimisation step.
+
+        args:
+            queries (List): List of tensors containing the encoded queries, shape [query_length]
+            responses (List): List of tensors containing the encoded responses, shape [response_length]
+            scores (List): tensor containing the scores, shape [batch_size]
+
+        returns:
+            train_stats (dict): a summary of the training statistics
+        """
+
+        bs = self.config['batch_size']
+        assert bs == len(queries), f"Batch size ({bs}) does not match number of examples ({len(queries)})"
+
+        timing = dict()
+        t0 = time.time()
+
+        response_lengths = [len(r) for r in responses]
+
+        t = time.time()
+        queries = [q.to(self.accelerator.device) for q in queries]
+        responses = [r.to(self.accelerator.device) for r in responses]
+        logprobs, ref_logprobs, values = self.batched_forward_pass(queries, responses)
+        timing['time/ppo/forward_pass'] = time.time()-t
+
+        t = time.time()
+        rewards, non_score_reward = self.compute_rewards(scores, logprobs, ref_logprobs)
+        timing['time/ppo/compute_rewards'] = time.time()-t
+
+        t = time.time()
+        all_stats = []
+        idxs = list(range(bs))
+        for _ in range(self.config['ppo_epochs']):
+            random.shuffle(idxs)
+            for i in range(bs):
+                idx = idxs[i]
+                train_stats = self.train_minibatch(logprobs[idx].unsqueeze(0), values[idx].unsqueeze(0),
+                                                   rewards[idx].unsqueeze(0), queries[idx].unsqueeze(0),
+                                                   responses[idx].unsqueeze(0),
+                                                   torch.cat([queries[idx],responses[idx]]).unsqueeze(0))
+                all_stats.append(train_stats)
+        timing['time/ppo/optimize_step'] = time.time()-t
+
+        t = time.time()
+        train_stats = stack_dicts(all_stats)
+
+        # reshape advantages/ratios such that they are not averaged.
+        train_stats['policy/advantages'] = torch.flatten(train_stats['policy/advantages']).unsqueeze(0)
+        train_stats['policy/advantages'] = torch.nan_to_num(train_stats['policy/advantages'], WANDB_PADDING)
+        train_stats['policy/ratio'] = torch.flatten(train_stats['policy/ratio']).unsqueeze(0)
+
+        stats = self.record_step_stats(scores=scores, logprobs=logprobs, ref_logprobs=ref_logprobs,
+                                       non_score_reward=non_score_reward, train_stats=train_stats,
+                                       kl_coef=self.kl_ctl.value)
+        stats = stats_to_np(stats)
+        timing['time/ppo/calc_stats'] = time.time()-t
+
+        self.kl_ctl.update(stats['objective/kl'], self.config['batch_size'])
+
+        timing['time/ppo/total'] = time.time()-t0
+        stats.update(timing)
+        return stats
```

Comment:
since you initialize the W&B logger with accelerate we can probably just pass all the stats to it internally and don't put the burden on the user.

Also, i think we can make the type of logger a kwarg that's passed to PPO so we don't require users to use W&B and they could also setup other loggers.